x-pack/metricbeat/module/aws/billing/billing.go

```diff
@@ -351,27 +360,32 @@ func constructMetricQueries(listMetricsOutput []types.Metric, dataGranularity ti
 	return metricDataQueries
 }
 
-func createMetricDataQuery(metric types.Metric, index int, dataGranularity time.Duration) types.MetricDataQuery {
+func createMetricDataQuery(metric aws.MetricWithID, index int, dataGranularity time.Duration) types.MetricDataQuery {
 	statistic := "Maximum"
 	dataGranularityInSeconds := int32(dataGranularity.Seconds())
 	id := metricsetName + strconv.Itoa(index)
-	metricDims := metric.Dimensions
-	metricName := *metric.MetricName
+	metricDims := metric.Metric.Dimensions
+	metricName := *metric.Metric.MetricName
 
-	label := metricName + labelSeparator
+	label := metric.AccountID + labelSeparator + "${PROP('AccountLabel')}" + labelSeparator + metricName + labelSeparator
```

Comment:
nit not related to this PR - it would be better if we could avoid the manual work of keeping this label in sync with the index constants used to parse it. one way would just be to wrap the label creation and parsing in a type; at least that way the indexing concerns don't bleed out into the code. let's not change it now, just a thought though.